src/json-schema.ts

```diff
@@ -0,0 +1,857 @@
+ * @returns True if object has the string property
+ */
+function hasStringProperty(
+    obj: unknown,
+    prop: string,
+): obj is Record<string, string> {
+    return isRecord(obj) && typeof obj[prop] === 'string';
+}
+
+/**
+ * Type guard to check if a value is a JsonSchema
+ *
+ * @param value - The value to check
+ * @returns True if the value is a JsonSchema object
+ */
+function isJsonSchema(value: unknown): value is JsonSchema {
+    return typeof value === 'object' && value !== null;
+}
+
+/**
+ * JSON Schema format mappings to Faker.js methods with proper typing
+ */
+const FORMAT_MAPPINGS: Record<string, (faker: Faker) => string> = {
+    'date': (faker: Faker): string =>
+        faker.date.recent().toISOString().split('T')[0],
+    'date-time': (faker: Faker): string => faker.date.recent().toISOString(),
+    'email': (faker: Faker): string => faker.internet.email(),
+    'hostname': (faker: Faker): string => faker.internet.domainName(),
+    'ipv4': (faker: Faker): string => faker.internet.ipv4(),
+    'ipv6': (faker: Faker): string => faker.internet.ipv6(),
+    'password': (faker: Faker): string => faker.internet.password(),
+    'regex': (faker: Faker): string => faker.lorem.word(),
+    'time': (faker: Faker): string =>
+        faker.date.recent().toTimeString().split(' ')[0],
+    'uri': (faker: Faker): string => faker.internet.url(),
+    'uri-reference': (faker: Faker): string => faker.internet.url(),
+    'url': (faker: Faker): string => faker.internet.url(),
+    'uuid': (faker: Faker): string => faker.string.uuid(),
+} as const;
+
+/**
+ * Factory class for creating JSON Schema-based factories
+ */
+export class JsonSchemaFactory<T> extends Factory<T> {
+    private readonly jsonSchemaOptions: JsonSchemaOptions;
+    private readonly schema: unknown;
+
+    private constructor(
+        schema: unknown,
+        options: JsonSchemaOptions,
+        validator: null | ValidateFunction<T>,
+    ) {
+        super(
+            (faker) => {
+                let generated: unknown;
+
+                // Handle root-level schemas by type
+                if (isJsonSchema(schema) && schema.type === 'array') {
+                    generated = generateArray(schema, faker, options, 0);
+                } else if (isJsonSchema(schema) && schema.type === 'null') {
+                    generated = null;
+                } else {
+                    generated = createValueGenerator(schema, faker, options, 0);
+                }
+
+                // Validate if strict validation is enabled
+                if (validator && options.strictValidation) {
+                    const validationResult = validator(generated);
+                    const isValid =
+                        typeof validationResult === 'boolean'
+                            ? validationResult
+                            : Boolean(validationResult);
+                    if (!isValid) {
+                        console.warn(
```

Comment:
no console.warn, we should throw. factory should always generate successful object complying with schema